extensions/wikia/SpecialManageWikiaHome/ManageWikiaHomeController.class.php

```diff
@@ -452,6 +452,10 @@ public function isWikiInCollection() {
 	public function changeFlag() {
 		wfProfileIn(__METHOD__);
 
+		if ( !$this->request->isValidWriteRequest( $this->getContext()->getUser() ) ) {
```

Comment:
`WikiaRequest::isValidWriteRequest` throws an exception, it doesn't return a boolean. Also, since you're in a controller, you can just call `$this->checkWriteRequest()` which will handle the user for you.